src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1028,7 +1028,9 @@ void PDP::check_remote_participant_liveliness(
                 std::chrono::microseconds(TimeConv::Duration_t2MicroSecondsInt64(remote_participant->m_leaseDuration));
         if (now > real_lease_tm)
         {
+            guard.unlock();
             remove_remote_participant(remote_participant->m_guid, ParticipantDiscoveryInfo::DROPPED_PARTICIPANT);
+            guard.lock();
```

Comment:
Not necessary to take again the mutex because the function will return immediately.

Reply:
@richiware can you do the necessary edit through the web interface?

Reply:
Done